### ISSUE TITLE
calamares/siduction install hangs at 93% - bootloader installation

### ISSUE THREAD

**Issue by btreut** [calamares/calamares#912]
#### Submission type

  - [ X] Bug report

#### Info regarding which version of Calamares is used, which Distribution

> Calamares 3.2.0rc2   Siduction 2018.1

#### Provide information on how the disks are set up, in detail, with full logs of commands issued 

> I am trying to set up a bootable USB stick on/for an Acer Laptop B118 with endlessOS preinstalled on
the SSD /dev/sda, which is partitioned as follows:

GPT fdisk (gdisk) version 1.0.1

Partition table scan:
  MBR: protective
  BSD: not present
  APM: not present
  GPT: present

Found valid GPT with protective MBR; using GPT.
Disk /dev/sda: 250069680 sectors, 119.2 GiB
Logical sector size: 512 bytes
Disk identifier (GUID): A66D3248-98FF-4A8B-AF9D-D68931C9CD08
Partition table holds up to 128 entries
First usable sector is 34, last usable sector is 250069646
Partitions will be aligned on 2048-sector boundaries
Total free space is 2669 sectors (1.3 MiB)

Number  Start (sector)    End (sector)  Size       Code  Name
   1            2048          129023   62.0 MiB    EF00  
   2          129024          131071   1024.0 KiB  EF02  
   3          131072       244006911   116.3 GiB   8304  
   4       244006912       250068991   2.9 GiB     ED01  Basic data partition

I prepared the 64GiB USB stick following these instructions: http://www.slivermetal.org/2016/09/18/how-to-create-an-hybrid-uefi-gpt-bios-gptmbr-boot-usb-disk/

GPT fdisk (gdisk) version 1.0.1

Partition table scan:
  MBR: protective
  BSD: not present
  APM: not present
  GPT: present

Found valid GPT with protective MBR; using GPT.
Disk /dev/sdb: 120127488 sectors, 57.3 GiB
Logical sector size: 512 bytes
Disk identifier (GUID): 8F4C718D-B405-4141-8074-591B07DA2026
Partition table holds up to 128 entries
First usable sector is 34, last usable sector is 120127454
Partitions will be aligned on 2048-sector boundaries
Total free space is 2014 sectors (1007.0 KiB)

Number  Start (sector)    End (sector)  Size       Code  Name
   1            2048            4095   1024.0 KiB  EF02  BIOS boot partition
   2            4096          106495   50.0 MiB    EF00  EFI System
   3          106496         2748415   1.3 GiB     0700  Microsoft basic data
   4         2748416       120127454   56.0 GiB    8300  

#### What do you expect to have happen when Calamares installs?

> a fully functional USB stick, which I have to select during boot by hitting F12 on acer splash screen and selecting the stick.

#### Describe the issue you encountered

> installation is stuck at 93% during attempt to install boot loader

#### Steps to reproduce the problem

> oops. Don't know on other systems, but with this Acer laptop, you have to prepare an bootable usb  stick with the siduction iso image dd copied. Disable secure boot (you have to temporarily enter a bios password to disable secure boot). Then boot the ISO-image stick and plug in the other USB stick.

#### Include the installation.log (usually ~/Calamares/Calamares/Calamares.log, of the user Calamares runs as):

> [calamares_install-d.log](https://github.com/calamares/calamares/files/1778412/calamares_install-d.log)

**Comment by agaida** [comment]
With my siduction hat on - why is this an upstream bug?

**Comment by kkofler** [comment]
This looks like the siduction live image is missing some of the files required to run `grub-install` on EFI.

The executable name `grub-install` should be correct on siduction. (Debian does not do the `grub2-*` rename that some other distros do.) So the command must be failing.

**Comment by agaida** [comment]
@kkofler: you are joking, right? :P

**Comment by kkofler** [comment]
Huh? No.

**Comment by btreut** [comment]
Sorry if this a siduction problem, but according to https://news.siduction.org an UEFI install  should be possible without problems.

**Comment by kkofler** [comment]
@agaida: Is this the longstanding Debian issue that the Debian packages for BIOS GRUB and GRUB-EFI conflict and so the image cannot ship both? I thought you folks had some solution for that?

**Comment by agaida** [comment]
@kkofler - long story short - there is no bug, not in siduction nor in debian. So this can be closed without further processing. The only thing we should do in calamares is to improve the stick handling. Right now it is nuts. (But this can be discussed and is worth a discussion - but is also not a bug at all. I would call it a 'handling problem'.

@btreut - and we should agree that downstream problems should be discussed downstream first. It might help a lot because downstream should know the implementation and known pitfalls. The pitfall was in that special case:

> I prepared the 64GiB USB stick following these instructions: http://www.slivermetal.org/2016/09/18/how-to-create-an-hybrid-uefi-gpt-bios-gptmbr-boot-usb-disk/

**Comment by agaida** [comment]
Tested several ways - and really found an error in my assumtions - my error was that i thought that calamares don't recognise a stick as possible installation target if it is the stick that contain the installation iso - that is plain wrong. Only sticks that are detected as iso9??? are excluded as installation target. Not nice, should be included in the documentation. So - beside of that there are no major bugs in the stick handling, but that not make sure that a installation has success.

Steps to install siduction on a stick - a will take the stick which contain the install medium:
1. Choose an iso - one can download from our mirror master: https://sync.siduction.org/iso/patience
2. Check the checksums of the downloaded iso
3. Put the iso onto a stick (cp, cat, dd - all is possible). I would suggest somthing like that:
```
dd bs=32M oflag=sync status=progress if=/home/agaida/Downloads/siduction-18.1.0-patience-lxqt-amd64-201712310045.iso of=/dev/sdg
```
4. Boot from the stick - use the option field and enter: `toram` - that means that the complet iso would be loaded to ram - the stick will never be used after loading to ram.
5. Once the choosen desktop appear start `gparted` and create a new partition table on the stick.
6. Start Calamares and choose the stick as target - Erase the drive should work fine
7. Install siduction and have fun ...

**Comment by btreut** [comment]
Hi Al,

thanks for the detailed instructions in your last post. But let me ask two questions beforehand:

First:

> @btreut - and we should agree that downstream problems should be discussed downstream first. It might help a lot because downstream should know the implementation and known pitfalls. 

what do mean with downstream and upstream here?

Second: 

> The pitfall was in that special case:

>> I prepared the 64GiB USB stick following these instructions: http://www.slivermetal.org/2016/09/18/how-to-create-an-hybrid-uefi-gpt-bios-gptmbr-boot-usb-disk/

what is wrong with these instructions?  They are not that much different from the UEFI notes in the siduction wiki (https://wiki.siduction.de/index.php?title=UEFI-Installation_f%C3%BCr_Siduction and https://wiki.siduction.de/index.php?title=Probleme_mit_UEFI_l%C3%B6sen).

Now several questions to the last post:

What is wrong in creating one stick to boot the Installation medium from and install to yet another (separate) stick (except, probably Speed) ?

As far as I understand UEFI until now, I will need an EFI Partition on the stick, since I cannot modify the endlessOS grub configuration to include the stick as bootable medium. I don't have much experience with UEFI, but I have had a lot experience with grub legacy.

**Comment by adriaandegroot** [comment]
@btreut Calamares itself -- this bug tracker -- is "upstream", and Siduction is "downstream" because Siduction uses Calamares; Alf with his Siduction hat on is a downstream user. You are a user of Siduction, so you're a downstream-of-Siduction user. So I think the point is that this should be reported in the Siduction bug tracker **first**, to figure out if it is a Siduction problem (e.g. wrong configuration of Calamares) or a Calamares problem. In the latter case, a Siduction person can move the issue upstream, into this bug tracker.

@agaida 
> Only sticks that are detected as iso9??? are excluded as installation target. Not nice, should be included in the documentation.
There's a bunch of reasons a device might be excluded from the selection. Calamares tells you why a particular device is removed; the device with / on it (presumably the USB stick you booted from), and devices for which blkid outputs "iso9660" for the device or any of its partitions. The debug log explains each one. I'm open to PRs adding extra documentation about which devices are suitable, but take a look at the open issues about rethinking the installation targets beforehand.

**Comment by agaida** [comment]
@ade - i kow both cases - but for me the problem is also typical "downstream" - was testing yesterday (two sticks) - calamares shows only the ssd - so the second stick was iso 9* too - took several minutes to remember and fígure it out - second thing - a really fine experience was to start with the "toram" parameter and install on the same stick. But it would be nice to have the "blocked" devices listed as blocked with the reason. Don't expect a user use the debug mode :)

@btreut - installation on a second stick is also possible - i called your way "pitfall" because it is easy to make mistakes within the prep or the target stick. Thats why i would suggest just to use the standard calamares functionality. It works - so based on this tests one can try to prepare a stick layout that fit his needs better.

**Comment by btreut** [comment]
@adriaandegroot  -

thanks for the clarification.

> In the latter case, a Siduction person can move the issue upstream, into this bug tracker.

should I move ongoing discussion to the siduction forum?

@agaida -

>Thats why i would suggest just to use the standard calamares functionality. 

as I understand it, the standard calamares functionality tries to modify the EFI config on sda, which does not make sense for endless, see: https://community.endlessos.com/t/dual-boot-endless-os-with-ubuntu/5569/4 so it is necessary to have a second EFI Partition on the stick.

**Comment by agaida** [comment]
TL;DR - and you understand it wrong
